vnext/ReactUWP/Modules/DeviceInfoModule.cpp

```diff
@@ -20,25 +20,27 @@ DeviceInfo::DeviceInfo(const std::shared_ptr<IReactInstance> &reactInstance) : m
 
 void DeviceInfo::update() {
   auto displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+  m_scale = static_cast<float>(displayInfo.ResolutionScale()) / 100;
   winrt::Windows::UI::ViewManagement::UISettings uiSettings;
 
   auto const &window = winrt::Windows::UI::Xaml::Window::Current().CoreWindow();
-
+  m_width = window.Bounds().Width;
+  m_height = window.Bounds().Height;
   m_dimensions = folly::dynamic::object(
       "windowPhysicalPixels",
-      folly::dynamic::object("width", window.Bounds().Width)("height", window.Bounds().Height)(
-          "scale", static_cast<int>(displayInfo.ResolutionScale()) / 100)("fontScale", uiSettings.TextScaleFactor())(
-          "densityDpi", displayInfo.LogicalDpi()))(
+      folly::dynamic::object("width", m_width * m_scale)("height", m_height * m_scale)("scale", m_scale)(
+          "fontScale", uiSettings.TextScaleFactor())("densityDpi", displayInfo.LogicalDpi()))(
       "screenPhysicalPixels",
       folly::dynamic::object("width", displayInfo.ScreenWidthInRawPixels())(
-          "height", displayInfo.ScreenHeightInRawPixels())(
-          "scale", static_cast<int>(displayInfo.ResolutionScale()) / 100)("fontScale", uiSettings.TextScaleFactor())(
+          "height", displayInfo.ScreenHeightInRawPixels())("scale", m_scale)("fontScale", uiSettings.TextScaleFactor())(
           "densityDpi", displayInfo.LogicalDpi()));
 }
 
 void DeviceInfo::updateRootElementSize(float width, float height) {
-  m_dimensions["windowPhysicalPixels"]["width"] = width;
-  m_dimensions["windowPhysicalPixels"]["height"] = height;
+  m_width = width;
+  m_height = height;
+  m_dimensions["windowPhysicalPixels"]["width"] = width * m_scale;
```

Comment:
>m_scale [](start = 58, length = 7)

I no longer see any code responding to DPI changes.  If you change the scale factor, then change the window size, it seems like this code will continue to use the old scale factor which would be a bug.

Reply:
Making sure DPI changed event is handled in my latest iteration

---
In reply to: [372055798](https://github.com/microsoft/react-native-windows/pull/3956#discussion_r372055798) [](ancestors = 372055798)